Tools/upm_package.py

```diff
@@ -20,45 +20,65 @@ def main():
     parser.add_argument("-o", "--output", help="Output location, will use default build/npm location if unspecified", type=str.lower)
     parser.add_argument("-v", "--version", help="Semantic version string for the package", type=str.lower)
     parser.add_argument("-p", "--publish", help="Publish the package to NPM feed", action='store_true')
+    parser.add_argument("-t", "--tarball", help="Publish specified tarball", type=str.lower)
+    parser.add_argument("-d", "--dryrun", help="Simulate publishing without pushing", action='store_true')
     args = parser.parse_args()
 
-    # Copy plugin binaries to project location
-    if not args.stage:
-        stage.stage_binaries()
-    else:
-        stage.stage_binaries(args.stage)
+    if not args.tarball:

```

Comment:
Is there a reason why you start with !tarball instead of checking for args=tarball?  It's easier for me to think in the positive case instead of negating the comparator.

Reply:
Well, the script takes the plugin folder into a tarball which then gets pushed to the feed or it can just pack it as a tarball to be pushed later. The initial idea was to push to the internal feed during the packaging stage and store the tarball as an artifact which can be then downloaded during the "publish" stage to be pushed to the public feed. Unfortunately pushing the tarball is cryptically failing on the CI build machine. Long winded way to say this wasn't super intentional. :)